# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/media-conduit/VideoConduit.cpp
# Commit: 0fc8c65e163e
# Full Hash: 0fc8c65e163ed5ccfc09c55ec906973cf35a79b4
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2018-11-03 09:44:53
# Regressor Bug: 1376873
# File Overlap Count: 3
# Description:
#   Bug 1376873 - Use audio/video sync groups; r=bwc
#   
#   This uses sync groups in the receive stream configs for the conduits rather
#   than establishing sync through direct calls. When the streams are created
#   in call.cc, ConfigureSync is called, which results in SetSync being called
# ==============================================================================

diff -r de3f5bd6846b -r 0fc8c65e163e media/webrtc/signaling/src/media-conduit/VideoConduit.cpp
--- a/media/webrtc/signaling/src/media-conduit/VideoConduit.cpp	Wed Feb 07 15:00:17 2018 -0500
+++ b/media/webrtc/signaling/src/media-conduit/VideoConduit.cpp	Fri Apr 27 07:25:19 2018 -0400
@@ -493,7 +493,6 @@
   mCall->UnregisterConduit(this);
 
   // Release AudioConduit first by dropping reference on MainThread, where it expects to be
-  SyncTo(nullptr);
   MOZ_ASSERT(!mSendStream && !mRecvStream, "Call DeleteStreams prior to ~WebrtcVideoConduit.");
 }
 
@@ -570,6 +569,11 @@
   return true;
 }
 
+void WebrtcVideoConduit::SetSyncGroup(const std::string& group)
+{
+  mRecvStreamConfig.sync_group = group;
+}
+
 MediaConduitErrorCode
 WebrtcVideoConduit::ConfigureCodecMode(webrtc::VideoCodecMode mode)
 {
@@ -1420,31 +1424,6 @@
   DeleteRecvStream();
 }
 
-void
-WebrtcVideoConduit::SyncTo(WebrtcAudioConduit* aConduit)
-{
-  MOZ_ASSERT(NS_IsMainThread());
-
-  CSFLogDebug(LOGTAG, "%s Synced to %p", __FUNCTION__, aConduit);
-
-  if (!mRecvStream) {
-    CSFLogError(LOGTAG, "SyncTo called with no receive stream");
-    return;
-  }
-
-  MutexAutoLock lock(mMutex);
-  //TODO: fix sync in a post voice engine world
-/*
-  if (aConduit) {
-    mRecvStream->SetSyncChannel(aConduit->GetVoiceEngine(),
-                                aConduit->GetChannel());
-  } else if (mSyncedTo) {
-    mRecvStream->SetSyncChannel(mSyncedTo->GetVoiceEngine(), -1);
-  }
-*/
-  mSyncedTo = aConduit;
-}
-
 MediaConduitErrorCode
 WebrtcVideoConduit::AttachRenderer(RefPtr<mozilla::VideoRenderer> aVideoRenderer)
 {